Mathlib/Analysis/SpecialFunctions/Bernstein.lean

```diff
@@ -209,63 +168,81 @@ for a continuous function `f : C([0,1], ℝ)` converge uniformly to `f` as `n` t
 This is the proof given in [Richard Beals' *Analysis, an introduction*][beals-analysis], §7D,
 and reproduced on wikipedia.
 -/
-theorem bernsteinApproximation_uniform [NormedSpace ℝ E] (f : C(I, E)) :
+theorem bernsteinApproximation_uniform [LocallyConvexSpace ℝ E] (f : C(I, E)) :
     Tendsto (fun n : ℕ => bernsteinApproximation n f) atTop (𝓝 f) := by
-  simp only [Metric.nhds_basis_ball.tendsto_right_iff, Metric.mem_ball, dist_eq_norm]
-  intro ε h
-  let δ := δ f ε h
-  have nhds_zero := tendsto_const_div_atTop_nhds_zero_nat (2 * ‖f‖ * δ ^ (-2 : ℤ))
-  filter_upwards [nhds_zero.eventually_lt_const (half_pos h), eventually_ne_atTop 0]
-    with n nh hn₀
-  -- As `[0,1]` is compact, it suffices to check the inequality pointwise.
-  rw [ContinuousMap.norm_lt_iff _ h]
-  intro x
-  -- The idea is to split up the sum over `k` into two sets,
-  -- `S`, where `x - k/n < δ`, and its complement.
```

Comment:
Could you keep this comment somewhere?

Also, since locally convex spaces are less standard than normed spaces, could you comment about them in the proof, explaining that the reader may think that `gauge U` is like a norm?

Reply:
@sgouezel I've restored this comment and added more explanations. I'm going to merge this in 5-6 hours unless you'll request more comments.